ppr-ui/src/views/newMhrRegistration/HomeOwners.vue

```diff
@@ -121,68 +120,94 @@
-    SimpleHelpToggle
-  }
-})
-export default class HomeOwners extends Vue {
-  @Getter getMhrRegistrationHomeOwners: MhrRegistrationHomeOwnersIF[]
-  @Action setMhrRegistrationHomeOwners: ActionBindingIF
-
-  private showAddPersonSection = false
-  private showAddPersonOrganizationSection = false
-  private isEditingMode = false
-
-  private get tenancyType (): string {
-    if (this.getMhrRegistrationHomeOwners?.length === 0) return 'N/A'
-    return this.getMhrRegistrationHomeOwners?.length === 1
-      ? 'Sole Ownership'
-      : 'Joint Tenants'
-  }
-
-  private get disableAddHomeOwnerBtn (): boolean {
-    return (
-      this.showAddPersonOrganizationSection ||
-      this.showAddPersonSection ||
-      this.isEditingMode
+    HomeOwnersTable
+  },
+  setup () {
+    const { getMhrRegistrationHomeOwners } = useGetters<any>([
+      'getMhrRegistrationHomeOwners'
+    ])
+
+    const { setMhrRegistrationHomeOwners } = useActions<any>([
+      'setMhrRegistrationHomeOwners'
+    ])
+
+    const { setGlobalEditingMode } = useHomeOwners()
+
+    const localState = reactive({
+      showAddPersonSection: false,
+      showAddPersonOrganizationSection: false,
+      disableAddHomeOwnerBtn: computed(() => {
+        return (
+          localState.showAddPersonOrganizationSection ||
+          localState.showAddPersonSection
+        )
+      }),
+      tenancyType: computed(() => {
+        if (getMhrRegistrationHomeOwners.value?.length === 0) return 'N/A'
+        return getMhrRegistrationHomeOwners.value?.length === 1
+          ? 'Sole Ownership'
+          : 'Joint Tenants'
+      })
+    })
+
+    // Enable editing mode whenever adding Person or Business
+    // This would disabled all Edit buttons
+    watch(
+      () => localState.disableAddHomeOwnerBtn,
+      (isAdding: Boolean) => {
+        setGlobalEditingMode(isAdding)
+      }
     )
-  }
-
-  private async addHomeOwner (owner): Promise<void> {
-    const homeOwners = [...this.getMhrRegistrationHomeOwners]
-    homeOwners.push(owner)
-    this.setMhrRegistrationHomeOwners(homeOwners)
-  }
 
-  private async editHomeOwner (owner): Promise<void> {
-    const homeOwners = [...this.getMhrRegistrationHomeOwners]
-    const { id, ...editedOwner } = owner
-    homeOwners[owner.id] = editedOwner
-    this.setMhrRegistrationHomeOwners(homeOwners)
+    const addHomeOwner = owner => {
+      const homeOwners = [...getMhrRegistrationHomeOwners.value]
+      homeOwners.push(owner)
+      setMhrRegistrationHomeOwners(homeOwners)
+    }
+
+    const editHomeOwner = owner => {
+      const homeOwners = [...getMhrRegistrationHomeOwners.value]
+      const { id, ...editedOwner } = owner
+      homeOwners[owner.id] = editedOwner
+      setMhrRegistrationHomeOwners(homeOwners)
+    }
+
+    const removeHomeOwner = owner => {
+      const homeOwners = [...getMhrRegistrationHomeOwners.value]
+      homeOwners.splice(homeOwners.indexOf(owner), 1)
+      setMhrRegistrationHomeOwners(homeOwners)
+    }
```

Comment:
Not major but for consistency, might be a good idea to give an argument type here so we know what `owner` is

Reply:
Added types for owners.